src/main/java/org/gridsuite/filter/server/FilterService.java

```diff
@@ -549,6 +549,10 @@ private List<Identifiable<?>> getVoltageLevelList(Network network, AbstractFilte
             return network.getVoltageLevelStream()
                 .filter(voltageLevel -> equipmentIds.contains(voltageLevel.getId()))
                 .collect(Collectors.toList());
+        } else if (filter instanceof ExpertFilter expertFilter) {
+            Stream<Identifiable<?>> stream = network.getVoltageLevelStream().map(voltageLevel -> voltageLevel);
+            var rule = expertFilter.getRules();
+            return stream.filter(rule::evaluateRule).toList();
```

Comment:
```suggestion
var rule = expertFilter.getRules();
            return network.getVoltageLevelStream().map(voltageLevel -> (Identifiable<?>) voltageLevel).filter(rule::evaluateRule).collect(Collectors.toList());
```
this also seems to work although we have to use `collect(Collectors.toList())` otherwise there's a type inference issue, it's up to what you prefer

Reply:
Not a big deal, I can change it